tests/SimpleStringTest.cpp

```diff
@@ -257,6 +257,12 @@ TEST(SimpleString, countTogether)
     LONGS_EQUAL(4, str.count("ha"));
 }
 
+TEST(SimpleString, countEmptyString)
+{
+    SimpleString str("hahahaha");
+    LONGS_EQUAL(1, str.count(""));
```

Comment:
What would you rather expect? How many times is the empty string contained in a string? :P

Behavior so far was some very large unsigned long integer value, possibly MAX_ULONG. This lead to problems with the empty string in replace().